### DISCUSSION TITLE
Unable to write/save to vault when external to network

### DISCUSSION THREAD

**Discussion by cjramseyer** [hassio-addons/addon-bitwarden#237] [General]
I am self-hosting, and running the latest version of the add-on, along with the latest version of HA.
It runs very stable.  I have setup remote access using duckdns, and I have open the one port on my firewall.

I can access my vault remotely on multiple devices.

However, I cannot add NEW entries to my vault when external to my network.
This suggests a port or other issue required by a write operation that is not needed for read operations.

**Comment by beMoD** [comment]
Are you sure you can actually read the vault when on an external network? The client has a cache feature when offline so you have access to your whole vault even when offline but you cannot write while offline

**Reply by cjramseyer** [reply]
I have actually been able to solve this issue and determine it's root cause.  The addon does not use a valid certificate, so, while the app works, it give no indication that there is a cert problem.  So, because of the cert issue, when external the app seems to "decide" it is offline.

My solution?: Using cloudflare, I expose the service (without port forwarding or hole in firewall), present a proper cert, and enforce 2fa to use the app.  Now this is working perfectly.
